[Build/CI] Add tracing deps to vllm container image

russellb opened the pull request:
A report requested that these dependencies be included in the image
for tracing. That seems reasonable - they are needed by
`vllm.tracing`. Stop manually installing them in the relevant test
pipeline and just include them in the image.

Signed-off-by: Russell Bryant <rbryant@redhat.com>

russellb committed:
[Build/CI] Add tracing deps to vllm container image

A report requested that these dependencies be included in the image
for tracing. That seems reasonable - they are needed by
`vllm.tracing`. Stop manually installing them in the relevant test
pipeline and just include them in the image.

Signed-off-by: Russell Bryant <rbryant@redhat.com>

hmellor reviewed (approved):
LGTM!

Perhaps a comment saying why these deps are here, similar to other deps? Something like `# Required for vllm.tracing`

russellb commented:
> LGTM!
> 
> Perhaps a comment saying why these deps are here, similar to other deps? Something like `# Required for vllm.tracing`

Thanks! rebased and added those comments

DarkLight1337 commented:
Can you merge from main to fix docker build?

russellb commented:
> Can you merge from main to fix docker build?

done - rebased on main

robertaistleitner commented:
This actually broke my python dependencies since we're also using grpc + protobuf, but with protobuf version > 5, which has a dependency conflict with opentelemetry-exporter-otlp>=1.26.0,<1.27.0. In my opinion, this should be an optional dependency (pip extra maybe?).

If this is not an option, another possibility would be to upgrade the opentelemetry dependencies to at least 1.28.0 since protobuf v5 is used from there on (https://github.com/open-telemetry/opentelemetry-python/commit/d1904b9529fc096b9ad8c13febf26f0876f7cef4).

Looking forward to your response.

hmellor commented:
There will be a PR soon that removes the upper limit on the `opentelemetry-exporter-otlp` version, will that solve the problem for you?

robertaistleitner commented:
Yes this fixes the issue for me, temporarily fixed it in https://github.com/robertaistleitner/vllm/tree/stable/v0.8.5.post1 for my use-case using from source install with prebuilt wheel (`git clone` and `pip install -e .`) and setting env `VLLM_USE_PRECOMPILED=1` and `VLLM_PRECOMPILED_WHEEL_LOCATION=https://files.pythonhosted.org/packages/47/07/e8822ca825f88aec61a7af6210b85cd0e0a4d6f757081d3282f32b1b0912/vllm-0.8.5.post1-cp38-abi3-manylinux1_x86_64.whl`.
